crates/iota-framework/packages/iota-system/tests/rewards_distribution_tests.move

```diff
@@ -972,11 +974,137 @@ module iota_system::rewards_distribution_tests {
+
+        // Get exchange rates for all pools and check that their rates are constant during the last epoch change
+        scenario.next_tx(VALIDATOR_ADDR_1);
+        let mut system_state = scenario.take_shared<IotaSystemState>();
+
+        let staked_iota_1 = scenario.take_from_address<StakedIota>(STAKER_ADDR_1);
+        let pool_id_1 = staked_iota_1.pool_id();
+        let rates1 = system_state.pool_exchange_rates(&pool_id_1);
+        let pool_token_amount_before_1  = rates1[4].pool_token_amount() as u128;
+        let pool_token_amount_after_1 = rates1[5].pool_token_amount() as u128;
+        let iota_amount_before_1 = rates1[4].iota_amount() as u128;
+        let iota_amount_after_1 = rates1[5].iota_amount() as u128;
+        test_scenario::return_to_address(STAKER_ADDR_1, staked_iota_1);
+        assert!(iota_amount_after_1  * pool_token_amount_before_1 == pool_token_amount_after_1 * iota_amount_before_1, 0);
+
+        let staked_iota_2 = scenario.take_from_address<StakedIota>(STAKER_ADDR_2);
+        let pool_id_2 = staked_iota_2.pool_id();
+        let rates2 = system_state.pool_exchange_rates(&pool_id_2);
+        let pool_token_amount_before_2 = rates2[4].pool_token_amount() as u128;
+        let pool_token_amount_after_2 = rates2[5].pool_token_amount() as u128;
+        let iota_amount_before_2 = rates2[4].iota_amount() as u128;
+        let iota_amount_after_2 = rates2[5].iota_amount() as u128;
+        test_scenario::return_to_address(STAKER_ADDR_2, staked_iota_2);
+        assert!(iota_amount_after_2 * pool_token_amount_before_2 == pool_token_amount_after_2 * iota_amount_before_2, 0);
+
+        let staked_iota_3 = scenario.take_from_address<StakedIota>(STAKER_ADDR_4);
+        let pool_id_3 = staked_iota_3.pool_id();
+        let rates3 = system_state.pool_exchange_rates(&pool_id_3);
+        let pool_token_amount_before_3 = rates3[4].pool_token_amount() as u128;
+        let pool_token_amount_after_3 = rates3[5].pool_token_amount() as u128;
+        let iota_amount_before_3 = rates3[4].iota_amount() as u128;
+        let iota_amount_after_3 = rates3[5].iota_amount() as u128;
+        test_scenario::return_to_address(STAKER_ADDR_4, staked_iota_3);
+        assert!(iota_amount_after_3 * pool_token_amount_before_3 == pool_token_amount_after_3 * iota_amount_before_3, 0);
+
+        test_scenario::return_shared(system_state);
+        scenario_val.end();
+    }
+
+    #[test]
+    fun test_pool_tokens_minted() {
+        set_up_iota_system_state();
+        let mut scenario_val = test_scenario::begin(VALIDATOR_ADDR_1);
+        let scenario = &mut scenario_val;
+
+        // Add some stake to the pool
+        stake_with(STAKER_ADDR_1, VALIDATOR_ADDR_1, 1, scenario);
+        stake_with(STAKER_ADDR_2, VALIDATOR_ADDR_1, 50, scenario);
+
+        // need to advance epoch so staking starts counting
+        advance_epoch(scenario);
+       
+        // Advance a couple of epochs to make the exchange rate different than 1.
```

Comment:
Annotate with a comment stating the epoch number each time we call advance epoch. E.g. "// advance from epoch 3 to 4" 
This would be helpful in both tests

Reply:
done